Sofa/framework/Helper/src/sofa/helper/NameDecoder.h

```diff
@@ -264,6 +282,8 @@ std::string GetSofaTypeTemplateName(const std::string prefix)
 {
     if constexpr (HasName<T>::value )
             return prefix + T::Name();
+    if constexpr (HasDataTypeInfo<T>::value )
+            return prefix + sofa::defaulttype::DataTypeInfo<T>::name();
     return prefix + sofa::helper::NameDecoder::decodeTypeName(typeid(T));
```

Comment:
```suggestion
    if constexpr (HasName<T>::value )
    {
        return prefix + T::Name();
    }
    else if constexpr (HasDataTypeInfo<T>::value )
    {
        return prefix + sofa::defaulttype::DataTypeInfo<T>::name();
    }
    else
    {
        return prefix + sofa::helper::NameDecoder::decodeTypeName(typeid(T));
    }
```

This is to avoid multiple return statements in the same scope

Reply:
Interesting, never thought about that (but fixed)